### DISCUSSION TITLE
Back End Set Up Issues

### DISCUSSION THREAD

**Discussion by DevDeCoste** [Project-Books/book-project#870] [Q&A] [answered]
Hi all,

Trying to get everything running. I got the front end login to open in localhost no problem, but when trying to run the backend in IntelliJ I can't get it going with the right Run Config/it won't let me select BookProjectApplication.java as the main class to run.

As mentioned, I use IntelliJ IDEA Community, I have OpenJDK 15, as well as JDK 1.8.

I ran mvn clean install into the backend directory via command line terminal, book-app successful, email failed and the last one was skipped (I forget what the 3rd one was, sorry).

I have worked on a fairly extensive project using Maven, Spring Boot, Spring Data and Java, but did not use Docker or Yarn for that project. This is also my first time delving into Open Source contributions, so apologies for the newbie questions. :)
![runconfig](https://user-images.githubusercontent.com/8299280/150689517-589a9665-8b66-4dc2-a747-f77baf93ed25.png)
![jdks](https://user-images.githubusercontent.com/8299280/150689524-489acf9a-a8cf-4957-b665-d845de8a3588.png)
![backend](https://user-images.githubusercontent.com/8299280/150689545-7ab6497f-4bc7-4f61-adf3-9600dd7d723a.png)
![docker-backend](https://user-images.githubusercontent.com/8299280/150689627-4a439fd3-5354-49b7-9cad-5a22244d98d0.png)

**Comment by DevDeCoste** [comment, marked as answer]
Update: Was able to get them both running but the login still does not work. I tried connecting via Postman and got the error Connection Refused. Still not fully up and running yet.

**Reply by knjk04** [reply]
What endpoint are you trying with Postman and with what response body/params?

**Reply by DevDeCoste** [reply]
Attache
![cantconnect-postman](https://user-images.githubusercontent.com/8299280/151179327-e0f10b94-9ab6-4909-acd8-537ff6000310.png)
d

**Reply by knjk04** [reply]
That's strange. It should work when the backend and the database are running. 

One thing to check is that the username is correct (I think it should be, but verify by checking in the `user` table in the database). 

You can also check to see the logs on the backend

**Reply by DevDeCoste** [reply]
How do you recommend I check the database? None of the logins will work for me, though I am connected to the backend and can run the frontend. Also where in the backend docker logs would I find what I need?

Sorry, these are new technologies for me, I'm learning.

**Reply by knjk04** [reply]
If you started the phpmyadmin container, you can follow these instructions: https://github.com/Project-Books/book-project#access-database 

If that doesn't work, could you please let us know the exact steps you followed to start everything and send a GitHub gist containing all logs so that we can investigate by looking to reproduce the problem you are facing?

**Reply by knjk04** [reply]
@DevDeCoste Are you still facing this issue?

**Reply by DevDeCoste** [reply]
Yes, Im working with a developer friends trying to problem solve but as of now I still cant login to the front end with both front and backend connected, nor can I connect to the DB with provided credentials. DB says that my IP is not allowed access.

**Reply by DevDeCoste** [reply]
The process I'm following is via IntelliJ Community Edition
1. Open terminal for project and navigate to cd frontend of the book project
2. run docker-compose up --build
3. Open new terminal and cd to backend
4. run docker-compose up -d mysql phpmyadmin
5. Run project locally via IntelliJ
6. Attempt login to frontend at http://localhost:3000/. Get red "Sorry, something went wrong on our end. Please try again later." error
7. Attempt to login to database at http://localhost:8081/, get "IP not allowed to access database error"
8. Attempt to connect via Postman, get same "Error: connect ECONNREFUSED 127.0.0.1:8080" error

Doing the above has both the frontend and backend running in docker but everything is erroring out. My engineer friend who took a look also ran into all the same issues I did trying to get it to run locally on his machine. He is wondering if it is maybe how the database is configured but we are not sure.

**Reply by knjk04** [reply]
> Get red "Sorry, something went wrong on our end. Please try again later." error

That means something went wrong on the backend. 

We have now migrated from MySQL to PostgreSQL, so could you please try again with the new instructions on the README? If the issue persists, please let me know and please paste all output from the database docker output and the backend output (console window in IntelliJ) into a [GitHub Gist](https://gist.github.com/). 

I'm looking to reproduce this on my end (no luck so far), but it is hard without seeing the full log output in a GitHub gist

**Reply by DevDeCoste** [reply]
The new configuration works! I was able to get everything launched and logged in w/ Docker & command line. Thanks for your help!

**Reply by knjk04** [reply]
That's great to hear!

**Comment by DevDeCoste** [comment]
This is new.. There were only 2 backend services running, now there's a third that wont start up
![unable-to-access](https://user-images.githubusercontent.com/8299280/151430954-a864deaa-29eb-48bb-a30b-e648496626d0.png)
p

**Reply by knjk04** [reply]
Thanks for bringing this up.

We don't currently have instructions for running the backend with Docker on our README: https://github.com/Project-Books/book-project#running-the-backend. For now, you can try running the app in a Java IDE. (If you use Maven, you may encounter issue #893).

The issue is fairly new. The jar we need to run is in the book-app directory. Related PR: #910. Related issue: #909. Once #893 is fixed, I'm hoping PR #910 should work.
